clab/qemu/vm/launch.sh

```
@@ -0,0 +1,160 @@
+sudo chown "$(id -u):$(id -g)" "${SERIAL_LOG}" "${PID_FILE}" 2>/dev/null || true
+
+echo "QEMU started (PID $(cat "${PID_FILE}"))"
+
+# --- Wait for SSH ---
+echo "Waiting for VM to become SSH-reachable (port ${SSH_PORT})..."
+wait_for_ssh "after launch"
+echo "VM is SSH-reachable."
+
+# --- Wait for cloud-init ---
+echo "Waiting for cloud-init to complete..."
+ssh ${SSH_OPTS} -p "${SSH_PORT}" openperouter@localhost \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)